disperser/apiserver/server.go

```diff
@@ -558,21 +564,64 @@ func (s *DispersalServer) Start(ctx context.Context) error {
 	return nil
 }
 
-func (s *DispersalServer) updateQuorumCount(ctx context.Context) error {
+// updateQuorumConfig updates the quorum config and returns the updated quorum config. If the update fails,
+// it will fallback to the old quorumConfig if it is set. This is to improve the robustness of the disperser to
+// RPC failures since the quorum config is rarely updated. In the event that quorumConfig is incorrect, this will
+// not result in a safety failure since all parameters are separately validated on the smart contract.
+
+func (s *DispersalServer) updateQuorumConfig(ctx context.Context) (QuorumConfig, error) {
+
+	newConfig := s.quorumConfig
```

Comment:
Do we need a read lock here?
Actually, this value is getting updated throughout this entire method. Do we need to lock the entire method? That can slow down the requests as this portion of the requests are called sequentially

Reply:
I'm using a struct here instead of a pointer to a struct, so I think this line should be making a copy. 

But yeah, this probably needs a read lock here.